saleor/order/views.py

```diff
@@ -31,10 +31,10 @@ def details(request, token):
     orders = orders.select_related(
         'billing_address', 'shipping_address', 'user')
     order = get_object_or_404(orders, token=token)
-    notes = order.notes.filter(is_public=True)
+    user = request.user if request.user.is_authenticated else None
+    notes = order.notes.filter(user=user)
```

Comment:
I see some trouble here, let's say if staff user added some note, and his account was deleted after, then `OrderNote` he added will be preserved, but `user` will be set to `None`. This may cause some data leakage. 

I wonder if we shouldn't leave `is_public` on the `OrderNote` to determine if the note was added by the customer or not, without a possibility to create a public note from the dashboard 

Or shall we leave `OrderNote` to customers only, and introduce another way for staff to add notes about orders/ fulfillments etc. 
cc @maarcingebala @NyanKiyoshi

Reply:
I'd remove this line at all and disable rendering notes in the customer's view. As in the original issue - those notes should be rewritten so that they're only avaialble to staff users.

To keep the customer notes functionality, I would add a simple Textfield in Cart and in Order models. We would use that to store a one-time note from the customer and display it somewhere in the dashboard.

Reply:
Thanks for Your feedback! I have a couple of questions. Should these new notes be visible to the customer? Should these fields be added in this PR or it should be a separate PR?

Reply:
If users can add notes to their order I think they should also be able to see it later in the storefront's order details page. I would prefer to do those changes in this PR to keep the "order notes" funtionality available on `master`.